nautobot/virtualization/filters.py

```diff
@@ -233,8 +243,21 @@ class VMInterfaceFilterSet(BaseFilterSet, StatusModelFilterSetMixin, CustomField
     mac_address = MultiValueMACAddressFilter(
         label="MAC address",
     )
+    vlan_vid = MultiValueCharFilter(method="filter_vlan_vid", label="Assigned VLAN (VID)")
+    vlan = MultiValueCharFilter(method="filter_vlan", label="Assigned VLAN")
+    ip_address = MultiValueCharFilter(method="filter_ip_address", label="IP Address")
     tag = TagFilter()
 
+    def filter_vlan(self, queryset, name, value):
+        return queryset.filter(Q(untagged_vlan__in=value) | Q(tagged_vlans__in=value))
+
+    def filter_vlan_vid(self, queryset, name, value):
+        return queryset.filter(Q(untagged_vlan__vid__in=value) | Q(tagged_vlans__vid__in=value))
```

Comment:
Should we have distinct filters for `untagged_vlan`, `tagged_vlans`, `untagged_vlan_vid`, and `tagged_vlans_vid` in addition and/or instead?